distributed/s3.py

```diff
@@ -0,0 +1,34 @@
+import boto3
+from dask.imperative import Value
+from distributed import default_executor
+from toolz import curry
+
+DEFAULT_PAGE_LENGHT = 1000
+
+
+def get_list_of_summary_objects(bucket_name, prefix='', page_size=DEFAULT_PAGE_LENGHT):
+    s3 = boto3.resource('s3')
+    return [summary_object for summary_object in
+            s3.Bucket(bucket_name).objects.filter(Prefix=prefix).page_size(page_size)]
+
+
+def read_content_from_keys(bucket, key):
+    import boto3
+    s3 = boto3.resource('s3')
+    return s3.Object(bucket, key).get()['Body'].read()
```

Comment:
There are a couple of ways of how authentication can be handled. In my use case, we attached IAM roles to the instances to have S3 access. There is also ways to specify AWS keys as environment variables. Moreover, passing keys around to ec2 instances is not recommended and is probably not very safe. `boto` automatically picks up both IAM roles or environment variables. It seems we may be able to leave authentication to be handled externally to `distributed`.